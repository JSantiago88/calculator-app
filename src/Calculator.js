import React, { useState } from 'react';

import { ButtonTable } from './components/ButtonTable';
import { Display } from './components/Display';
import { getResult } from './helpers/getResult';

export const Calculator = () => {

    const [initialValue, setInitialValue] = useState([0]);

    const handleSelectButton = (e) => {
        const value = !isNaN(parseInt(e.target.innerText)) ?
            parseInt(e.target.innerText) :
            e.target.innerText;

        switch (value) { //step
            case '+':
                addOperator(value);
                break;
            case '-':
                addOperator(value);
                break;
            case 'x':
                addOperator(value);
                break;
            case '/':
                addOperator(value);
                break;
            case '=':
                setInitialValue(getResult(initialValue));
                break;
            case 'AC':
                setInitialValue([0]);
                break;
            case ',':
                addOperator(value);
                break;
            default:
                setInitialValue((cats) => {
                    return cats[0] === 0 ? [parseFloat(value)] : [...cats, parseFloat(value)];
                });
                break;
        }

    };

    const addOperator = (value) => {
        setInitialValue((cats) => {
            if (cats[cats.length - 1] === '+' ||
                cats[cats.length - 1] === '-' ||
                cats[cats.length - 1] === 'x' ||
                cats[cats.length - 1] === '/' ||
                cats[cats.length - 1] === ',') return [...cats];

            return cats[0] === 0 ||
                cats[0] === '+' ||
                cats[0] === '-' ||
                cats[0] === 'x' ||
                cats[0] === '/' ||
                cats[0] === ',' ? [0] : [...cats, value];
        });
    };

    return (
        <div className="calculator">
            <h2>Calculator</h2>
            <Display initialValue={initialValue} />
            <hr />
            <ButtonTable handleSelectButton={handleSelectButton} />
        </div>
    );

};
