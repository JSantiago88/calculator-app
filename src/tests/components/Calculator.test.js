import React from 'react';
import { shallow } from 'enzyme';

import Calculator from '../../Calculator';

describe('Test in <Calculator />', () => {

    const wrapper = shallow(<Calculator />);

    test('should Calculator the component correctly', () => {

        expect(wrapper).toMatchSnapshot();
    });


    test('should have a css class in the div', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('calculator')).toBe(true);
    });

});
