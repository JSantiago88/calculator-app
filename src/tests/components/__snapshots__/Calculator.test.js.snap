// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Calculator /> should Calculator the component correctly 1`] = `
<div
  className="calculator"
>
  <h2>
    Calculator
  </h2>
  <Display
    initialValue={
      Array [
        0,
      ]
    }
  />
  <hr />
  <ButtonTable
    handleSelectButton={[Function]}
  />
</div>
`;
